### PR TITLE
Deprecate `service-telemetry`

### PR DESCRIPTION
**Pull request by Dexterp37**
This adds a deprecation annotation to `Telemetry` and `TelemetryHolder` in addition to the README and changelog entries.

---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/service/telemetry/README.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 A generic library for sending telemetry pings from Android applications to Mozilla's telemetry service.
 
+## Deprecated
+This component is now deprecated: no further updates or fixes will be provided.
+Use [the Glean SDK](../glean) instead.
+This library will not be removed until all projects using start using the Glean SDK.
```
Comment (pocmo) [inline components/service/telemetry/README.md]: This reads wrong. Did you mean to write "using it"?

File: components/service/telemetry/README.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 A generic library for sending telemetry pings from Android applications to Mozilla's telemetry service.
 
+## Deprecated
+This component is now deprecated: no further updates or fixes will be provided.
```
Comment (pocmo) [inline components/service/telemetry/README.md]: nit: I'll provide a fix if something blows up completely. But I know what you mean. 🙃

File: components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java [hunk 3]
```diff
@@ -35,6 +35,10 @@
 import kotlin.Unit;
 import kotlin.jvm.functions.Function0;
 
+/**
+ * @deprecated The whole service-telemetry library is deprecated. Please use the Glean SDK instead.
+ */
+@Deprecated
```
Comment (pocmo) [inline components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java]: nit: Some tests may break since they are using a deprecated API now (We may need to propagate the annotation).

File: docs/changelog.md [hunk 4]
```diff
@@ -35,6 +35,9 @@ permalink: /changelog/
   * Added `CustomTabState.trustedOrigins` extension method to turn the verification state of a custom tab into a list of origins.
   * Added `WebAppHideToolbarFeature.onTrustedScopesChange` to change the trusted scopes after the feature is created.
 
+* **service-telemetry**
+  * This component is now deprecated. Please use the [Glean SDK](https://mozilla.github.io/glean/book/index.html) instead.
+
```
Comment (pocmo) [inline docs/changelog.md]: Let's also mention what you wrote in the README here: That we are not removing this until apps are migrated.

File: components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java [hunk 5]
```diff
@@ -35,6 +35,10 @@
 import kotlin.Unit;
 import kotlin.jvm.functions.Function0;
 
+/**
+ * @deprecated The whole service-telemetry library is deprecated. Please use the Glean SDK instead.
```
Comment (pocmo) [inline components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java]: Maybe also drop a link to Glean here.

File: components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java (hunk 3 above)
Comment (Dexterp37) [reply components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java]: Do you mean application tests? Nothing seems to be breaking on ci (luckily :D )

File: components/service/telemetry/README.md (hunk 2 above)
Comment (Dexterp37) [reply components/service/telemetry/README.md]: Hah, good point, yes. Let me rephrase this to be less scary and surprising!

File: components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java (hunk 3 above)
Comment (pocmo) [reply components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java]: Okay, then ignore! :)

Comment (pocmo) [reply components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java]: Ah, you are probably lucky because this is Java. Those checks are only enabled for Kotlin. :)

**Comment by Dexterp37** [comment]
bors retry
